### PR TITLE
Fix named export for example components

### PR DESCRIPTION
**Pull request by mendrew**
iss: #1281

Updated examples-loader to import component with named export
Named export of Basic button is now working

I tried suggestion `const name = name$0[name] || name$0.default || name$0;`
CounterButton component is now falling with an error:
```
Invariant Violation: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
Check the render method of `FunctionComponentWrapper`.

```
CoutnerButtons is falling

Placeholder from my previous [report](https://github.com/styleguidist/react-styleguidist/issues/1281#issuecomment-498433301) is now working.

### CODE REVIEW

**Comment by mendrew** [comment]
Ok, I finally get the idea why with this fix we have problems with other components from example.
The reason is simple but I've been debugging a lot till I get it.

### First example: CounterButton. 
Markdown code for CounterButton looks like this 
```js
import Button from '../Button'
let ref
;<div>
  <CounterButton ref={r => (ref = r)} />
  <Button size="small" onClick={() => ref.set(0)}>
    Reset
  </Button>
</div>
```

The error is about this import `import Button from '../Button'`. Button is no longer has default export in my PR and Button should be imported using named import.

### Second example: was tricked by it 
```jsx
import React from 'react'
import Button from 'rsg-example/components/Button'
import Placeholder from 'rsg-example/components/Placeholder'
;<Button>
  <Placeholder />
</Button>
```

In this case I had error with `import Button from 'rsg-example/components/Button'`, I thought it is absolutely different button, but then I looked at the styleguidist config and found this
https://github.com/mendrew/react-styleguidist/blob/1ed431d693af196c8c4d8f84839c60ac06274f2b/examples/basic/styleguide.config.js#L8

It's **the same** button component, and we should import it in the same way with named import.

**Comment by mendrew** [comment]
I thinks that ast bulder code update `const name = name$0[name] || name$0.default || name$0;` is enough.

I'm going to remove WIP code and update tests.

**Comment by mendrew** [comment]
I've finally updated the PR. It's ready for review.

I'm wondering does it make sense to add named export to some basic example code, just to have this case in examples?

**Comment by sapegin** [comment]
Haha, this all sounds very familiar! ;-) And thanks for fixing so many things in Styleguidist!

> I'm wondering does it make sense to add named export to some basic example code, just to have this case in examples?

Yeah, that would make sense — tests aren't always enough to make sure things are still working.

**Comment by mendrew** [comment]
> Haha, this all sounds very familiar! ;-) And thanks for fixing so many things in Styleguidist!

You are welcome! I found the code pretty interesting and clear. So it's a kind of pleasure to debug 🙂.   I haven't used it yet. But definitely will. And all this design system area looks inspiring.

> Yeah, that would make sense — tests aren't always enough to make sure things are still working.

I've updated the PR. Added such kind of import into RandomButton, so it likely hurt anyone who uses somehow example/basic components.
Did update for RadmonButton markdown to be explicit in a way we import this button.

Any other suggestions? @sapegin

File: src/loaders/examples-loader.js [hunk 1]
```diff
@@ -82,8 +82,11 @@ export default function examplesLoader(source) {
 				b.variableDeclaration('const', [
 					b.variableDeclarator(
 						b.identifier(name),
-						b.logicalExpression('||', b.identifier(`${name}$0.default`), b.identifier(`${name}$0`))
-					),
+						b.logicalExpression('||',
```
Comment (sapegin) [inline src/loaders/examples-loader.js]: Let's update a comment above to:

```
// const name = name$0[name] || name$0.default || name$0;
```

Comment (mendrew) [reply src/loaders/examples-loader.js]: Oh, you are right! I was thinking about it but forgot! Updated.

**Comment by sapegin** [comment]
Thanks! 🦄

**Pull request merged**